### ISSUE TITLE
code与paper在stage2中有一处setting不同

### ISSUE THREAD

**Issue by Shihao-Chen** [yayafengzi/LMM-HiMTok#4]
作者你好，感谢你十分有意义的开源工作，我想请问一下，我尝试用stage3微调后的mask vae的权重作为初始化去训练internvl3 1B（只在Refcoco/Refcoco+/Refcocog三个数据集上），但是在训练stage2的过程中，发现repeat了数据集很多次（总计6M左右）依然只能将llm loss下降到1.5左右，mask loss下降到0.7左右，可视化出来的分割效果为比较粗糙。我recheck了一下代码中的细节，发现stage2的setting有一处与论文中不同：“ def set_encoder_requires_grad(self,requires_grad=False):
if self.encoder is not None:
self.encoder.requires_grad_(requires_grad)
self.latent_tokens.requires_grad=requires_grad
self.quantize.requires_grad_(requires_grad)”该函数将self.latent_tokens和self.quantize都一起与self.encoder冻结了，和论文中stage2的setting不一致（论文的图3是放开的），想请问一下这一块的setting设置是哪一种呢

**Comment by yayafengzi** [comment]
你好！感谢你关注这个工作。你提到的quantizer部分，我自己训的时候这两种设置效果上没什么区别。如果你的交叉熵loss只能降到1.5的话那可能是llm的问题，我自己训的时候repeat数据集很多次的情况下交叉熵loss可以降到0.7。学习分割token对llm还是有难度的，可以尝试扩大scale。
